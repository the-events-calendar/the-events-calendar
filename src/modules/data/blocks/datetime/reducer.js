/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const { parseFormats, toTime } = momentUtil;

export const defaultStateToMetaMap = {
	start: '_EventStartDate',
	end: '_EventEndDate',
	dateTimeSeparator: '_EventDateTimeSeparator',
	timeRangeSeparator: '_EventTimeRangeSeparator',
	allDay: '_EventAllDay',
	timeZone: '_EventTimezone',
};

export const setInitialState = ( data ) => {
	const { meta } = data;

	Object.keys( defaultStateToMetaMap ).forEach( ( key ) => {
		const metaKey = defaultStateToMetaMap[ key ];
		if ( meta.hasOwnProperty( metaKey ) ) {
			DEFAULT_STATE[ key ] = meta[ metaKey ];
		}
	} );

	const { start, end } = DEFAULT_STATE;

	DEFAULT_STATE.startTimeInput = toTime( parseFormats( start ) );
	DEFAULT_STATE.endTimeInput = toTime( parseFormats( end ) );
	DEFAULT_STATE.naturalLanguageLabel = date.rangeToNaturalLanguage( start, end );
	DEFAULT_STATE.multiDay = ! momentUtil.isSameDay( momentUtil.toMoment( start ), momentUtil.toMoment( end ) );
};

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

const defaultStartDateTime = momentUtil.toDateTime( defaultStartMoment );
const defaultEndDateTime = momentUtil.toDateTime( defaultEndMoment );

export const DEFAULT_STATE = {
	start: defaultStartDateTime,
	end: defaultEndDateTime,
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguageLabel: date.rangeToNaturalLanguage( defaultStartDateTime, defaultEndDateTime ),
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: globals.timezone().timeZone ? globals.timezone().timeZone : date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguageLabel: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		default:
			return state;
	}
};
