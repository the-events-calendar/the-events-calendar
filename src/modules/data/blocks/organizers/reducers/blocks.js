/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_BLOCK_ORGANIZER:
			return {
				...state,
				[ action.payload.id ]: action.payload.organizer,
			};
		case types.REMOVE_BLOCK_ORGANIZER:
			return Object.keys( state ).reduce( ( newState, id ) => {
				if ( id === action.payload.id ) {
					return newState;
				}

				return {
					...newState,
					[ id ]: state[ id ],
				};
			}, {} );
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_BLOCK_ORGANIZER:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_BLOCK_ORGANIZER:
			return state.filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
