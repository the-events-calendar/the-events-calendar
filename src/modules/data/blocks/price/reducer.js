/**
 * Wordpress dependenciess
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { settings, priceSettings } from '@moderntribe/common/utils/globals';
import { string } from '@moderntribe/common/utils';
import * as types from './types';

const position = string.isTruthy( settings() && settings().reverseCurrencyPosition )
	? 'suffix'
	: 'prefix';

export const DEFAULT_STATE = {
	position: priceSettings() && priceSettings().defaultCurrencyPosition
		? priceSettings().defaultCurrencyPosition
		: position,
	symbol: priceSettings() && priceSettings().defaultCurrencySymbol
		? priceSettings().defaultCurrencySymbol
		: __( '$', 'the-events-calendar' ),
	code: priceSettings() && priceSettings().defaultCurrencyCode
		? priceSettings().defaultCurrencyCode
		: __( 'USD', 'the-events-calendar' ),
	cost: '',
};

export const defaultStateToMetaMap = {
	position: '_EventCurrencyPosition',
	symbol: '_EventCurrencySymbol',
	code: '_EventCurrencyCode',
	cost: '_EventCost',
};

export const setInitialState = ( data ) => {
	const { meta } = data;

	Object.keys( defaultStateToMetaMap ).forEach( ( key ) => {
		const metaKey = defaultStateToMetaMap[ key ];
		if ( meta.hasOwnProperty( metaKey ) ) {
			DEFAULT_STATE[ key ] = meta[ metaKey ];
		}
	} );
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_PRICE_COST:
			return {
				...state,
				cost: action.payload.cost,
			};
		case types.SET_PRICE_POSITION:
			return {
				...state,
				position: action.payload.position,
			};
		case types.SET_PRICE_SYMBOL:
			return {
				...state,
				symbol: action.payload.symbol,
			};
		case types.SET_PRICE_CODE:
			return {
				...state,
				code: action.payload.code,
			};
		default:
			return state;
	}
};
