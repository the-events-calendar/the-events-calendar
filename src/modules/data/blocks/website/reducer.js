/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	url: '',
};

export const defaultStateToMetaMap = {
	url: '_EventURL',
};

export const setInitialState = ( data ) => {
	const { meta } = data;

	Object.keys( defaultStateToMetaMap ).forEach( ( key ) => {
		const metaKey = defaultStateToMetaMap[ key ];
		if ( meta.hasOwnProperty( metaKey ) ) {
			DEFAULT_STATE[ key ] = meta[ metaKey ];
		}
	} );
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_WEBSITE_URL:
			return {
				...state,
				url: action.payload.url,
			};
		default:
			return state;
	}
};
