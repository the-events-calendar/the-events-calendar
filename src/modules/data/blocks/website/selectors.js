/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const getWebsiteBlock = ( state ) => state.events.blocks.website;

export const getUrl = createSelector( [ getWebsiteBlock ], ( website ) => website.url );
