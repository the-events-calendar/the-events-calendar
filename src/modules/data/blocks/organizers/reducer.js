/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import { classic, classicSetInitialState, blocks } from './reducers';

export const setInitialState = ( data ) => {
	classicSetInitialState( data );
};

export default combineReducers( {
	blocks,
	classic,
} );
