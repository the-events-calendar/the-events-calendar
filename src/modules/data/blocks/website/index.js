/**
 * Internal dependencies
 */
import * as reducer from './reducer';
import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';

export default reducer.default;
export { reducer, selectors, actions, types };
