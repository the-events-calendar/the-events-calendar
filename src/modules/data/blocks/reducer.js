/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime, { reducer as datetimeReducer } from './datetime';
import organizers, { reducer as organizersReducer } from './organizers';
import price, { reducer as priceReducer } from './price';
import website, { reducer as websiteReducer } from './website';
import venue, { reducer as venueReducer } from './venue';

export const setInitialState = ( data ) => {
	datetimeReducer.setInitialState( data );
	organizersReducer.setInitialState( data );
	priceReducer.setInitialState( data );
	websiteReducer.setInitialState( data );
	venueReducer.setInitialState( data );
};

export default combineReducers( {
	datetime,
	venue,
	organizers,
	price,
	website,
} );
