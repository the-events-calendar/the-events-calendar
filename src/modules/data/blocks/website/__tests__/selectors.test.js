/**
 * Internal dependencies
 */
import { selectors } from '@moderntribe/events/data/blocks/website';
import { DEFAULT_STATE } from '@moderntribe/events/data/blocks/website/reducer';

const state = {
	events: {
		blocks: {
			website: DEFAULT_STATE,
		},
	},
};

describe( '[STORE] - Website selectors', () => {
	it( 'Should return the website block', () => {
		expect( selectors.getWebsiteBlock( state ) ).toMatchSnapshot();
	} );

	it( 'Should return the website url', () => {
		expect( selectors.getUrl( state ) ).toMatchSnapshot();
	} );
} );
