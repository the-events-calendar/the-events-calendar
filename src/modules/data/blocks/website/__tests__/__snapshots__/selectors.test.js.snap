// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Website selectors Should return the website block 1`] = `
{
  "url": "",
}
`;

exports[`[STORE] - Website selectors Should return the website url 1`] = `""`;
